src/model/markercollection.js

```diff
@@ -78,28 +96,30 @@ export default class MarkerCollection {
 	 * set is different, the marker in collection is removed and then new marker is added. If the range was same, nothing
 	 * happens and `false` is returned.
 	 *
+	 * @protected
 	 * @fires module:engine/model/markercollection~MarkerCollection#event:add
 	 * @fires module:engine/model/markercollection~MarkerCollection#event:remove
 	 * @param {String|module:engine/model/markercollection~Marker} markerOrName Name of marker to add or Marker instance to update.
 	 * @param {module:engine/model/range~Range} range Marker range.
+	 * @param {Boolean} managedUsingOperations Specifies whether the marker is managed using operations.
 	 * @returns {module:engine/model/markercollection~Marker} `Marker` instance added to the collection.
 	 */
-	set( markerOrName, range ) {
+	_set( markerOrName, range, managedUsingOperations ) {
 		const markerName = markerOrName instanceof Marker ? markerOrName.name : markerOrName;
 		const oldMarker = this._markers.get( markerName );
 
 		if ( oldMarker ) {
 			const oldRange = oldMarker.getRange();
 
-			if ( oldRange.isEqual( range ) ) {
+			if ( oldRange.isEqual( range ) && managedUsingOperations === oldMarker.managedUsingOperations ) {
 				return oldMarker;
 			}
 
-			this.remove( markerName );
+			this._remove( markerName );
 		}
 
 		const liveRange = LiveRange.createFromRange( range );
-		const marker = new Marker( markerName, liveRange );
+		const marker = new Marker( markerName, liveRange, managedUsingOperations );
 
 		this._markers.set( markerName, marker );
 		this.fire( 'add:' + markerName, marker );
```

Comment:
I think that we should rename this to `set` because it is fired also when marker value changes.

Reply:
I'm thinking about the whole event concept used here. Is it useful? If you manage the `Marker`, then you have direct access to it. If you don't have the access, then you won't have access to the `Marker`'s name. There are zero use cases, for now, only tests.

Maybe `this.fire( 'add', marker );` would be better? Then you'd have the power to access all markers and filter needed marker or marker group.

Reply:
Hmm. Nevermind. I misunderstood this problem.